### PR TITLE
Extended tuning fixes

### PR DESCRIPTION
**Pull request by Arne-W**
This PR adapts the extended tuning setup view for use with arducopter 3.6 and improves the parameter handling.

### CODE REVIEW

File: src/ui/configuration/CopterPidConfig.cc [hunk 1]
```diff
@@ -155,119 +153,41 @@ CopterPidConfig::CopterPidConfig(QWidget *parent) : AP2ConfigWidget(parent)
     initConnections();
 }
 
-void CopterPidConfig::mapParamNamesToBox() {
-
-    // AC3.4+ param name compatibility
+void CopterPidConfig::mapParamNamesToBox()
+{
+    // version check
     ArduPilotMegaMAV* apmMav = static_cast<ArduPilotMegaMAV*>(m_uas);
```
Comment (Arne-W) [inline src/ui/configuration/CopterPidConfig.cc]: should be dynamic cast

File: src/ui/configuration/CopterPidConfig.cc [hunk 2]
```diff
@@ -155,119 +153,41 @@ CopterPidConfig::CopterPidConfig(QWidget *parent) : AP2ConfigWidget(parent)
...
-    m_nameToBoxMap[rate_rll_filt_hz] = ui.rateRollFiltHzSpinBox;
-    m_nameToBoxMap[rate_rll_imax] = ui.rateRollIMAXSpinBox;
-
-    m_nameToBoxMap[rate_pit_p] = ui.ratePitchPSpinBox;
-    m_nameToBoxMap[rate_pit_i] = ui.ratePitchISpinBox;
-    m_nameToBoxMap[rate_pit_d] = ui.ratePitchDSpinBox;
-    m_nameToBoxMap[rate_pit_filt_hz] = ui.ratePitchFiltHzSpinBox;
-    m_nameToBoxMap[rate_pit_imax] = ui.ratePitchIMAXSpinBox;
+    APMFirmwareVersion version = apmMav->getFirmwareVersion();
+    QLOG_DEBUG() << "PID Config is set up for " << version.versionString();
```
Comment (Arne-W) [inline src/ui/configuration/CopterPidConfig.cc]: better text "Extended Tuning ..."

File: src/ui/configuration/CopterPidConfig.cc [hunk 3]
```diff
@@ -155,119 +153,41 @@ CopterPidConfig::CopterPidConfig(QWidget *parent) : AP2ConfigWidget(parent)
...
-    m_nameToBoxMap["WPNAV_SPEED"] = ui.wpNavSpeedSpinBox;
-    m_nameToBoxMap["WPNAV_RADIUS"] = ui.wpNavRadiusSpinBox;
-    m_nameToBoxMap["WPNAV_SPEED_DN"] = ui.wpNavSpeedDownSpinBox;
-    m_nameToBoxMap["WPNAV_LOIT_SPEED"] = ui.wpNavLoiterSpeedSpinBox;
-    m_nameToBoxMap["WPNAV_SPEED_UP"] = ui.wpNavSpeedUpSpinBox;
+    if(version.majorNumber() == 3)
+    {
+        if(version.minorNumber() >= 6)
+        {
+            QLOG_DEBUG() << "Setting up for 3.6+";
```
Comment (Arne-W) [inline src/ui/configuration/CopterPidConfig.cc]: better text

File: src/ui/configuration/CopterPidConfig.cc [hunk 4]
```diff
@@ -155,119 +153,41 @@ CopterPidConfig::CopterPidConfig(QWidget *parent) : AP2ConfigWidget(parent)
...
+            setupPID_APM_36();
+        }
+        else if(version.minorNumber() >= 4)
+        {
+            QLOG_DEBUG() << "Setting up for 3.4+";
```
Comment (Arne-W) [inline src/ui/configuration/CopterPidConfig.cc]: better text

File: src/ui/configuration/CopterPidConfig.cc [hunk 5]
```diff
@@ -155,119 +153,41 @@ CopterPidConfig::CopterPidConfig(QWidget *parent) : AP2ConfigWidget(parent)
...
+        }
+    }
+    else
+    {
+        QLOG_DEBUG() << "Using default setting";
+        setupPID_Default();
+    }
 
     //m_nameToBoxMap["TUNE_HIGH"] = ui.ch6MaxSpinBox;
    // m_nameToBoxMap["TUNE_LOW"] = ui.ch6MinSpinBox;
```
Comment (Arne-W) [inline src/ui/configuration/CopterPidConfig.cc]: remove

File: src/ui/configuration/CopterPidConfig.cc (hunk 1 above)
Comment (lintendoCelix) [reply src/ui/configuration/CopterPidConfig.cc]: how to get these fixed code to be loaded in OSX? thx

Comment (Arne-W) [reply src/ui/configuration/CopterPidConfig.cc]: check this [](http://firmware.ardupilot.org/Tools/APMPlanner/daily/2019-03-08/) for the hot stuff
and this [](http://firmware.ardupilot.org/Tools/APMPlanner/) for the officials

**Pull request merged**